testing/matrix/linter/rules/roles/wildcards.go

```diff
@@ -0,0 +1,177 @@
+		"d8:prometheus-metrics-adapter:horizontal-pod-autoscaler-external-metrics",
+	},
+	"vertical-pod-autoscaler/templates/rbac-for-us.yaml": {
+		"d8:vertical-pod-autoscaler:controllers-reader",
+	},
+	"ingress-nginx/templates/kruise/rbac-for-us.yaml": {
+		"d8:ingress-nginx:kruise-role",
+	},
+	"cilium-hubble/templates/ui/rbac-for-us.yaml": {
+		"d8:cilium-hubble:ui:reader",
+	},
+	"okmeter/templates/rbac-for-us.yaml": {
+		"d8:okmeter",
+	},
+	"openvpn/templates/openvpn/rbac-for-us.yaml": {
+		"openvpn",
+	},
+	"upmeter/templates/upmeter-agent/rbac-for-us.yaml": {
+		"upmeter-agent",
+		"d8:upmeter:upmeter-agent",
+	},
+	"upmeter/templates/upmeter/rbac-for-us.yaml": {
+		"d8:upmeter:upmeter",
+	},
+	"documentation/templates/rbac-for-us.yaml": {
+		"documentation:leases-edit",
+	},
+	"delivery/templates/argocd/application-controller/rbac-for-us.yaml": {
+		"d8:delivery:argocd:application-controller",
+	},
+	"delivery/templates/argocd/server/rbac-for-us.yaml": {
+		"d8:delivery:argocd:server",
+	},
+	"cloud-provider-openstack/templates/cloud-controller-manager/rbac-for-us.yaml": {
+		"d8:cloud-provider-openstack:cloud-controller-manager",
+	},
+	"cloud-provider-vcd/templates/cloud-controller-manager/rbac-for-us.yaml": {
+		"d8:cloud-provider-vcd:cloud-controller-manager",
+	},
+	"cloud-provider-vsphere/templates/cloud-controller-manager/rbac-for-us.yaml": {
+		"d8:cloud-provider-vsphere:cloud-controller-manager",
+	},
+	"cloud-provider-zvirt/templates/cloud-controller-manager/rbac-for-us.yaml": {
+		"d8:cloud-provider-zvirt:cloud-controller-manager",
+	},
+}
+
+// ObjectRolesWildcard is a linter for checking the presence
+// of a wildcard in a Role and ClusterRole
+func ObjectRolesWildcard(object storage.StoreObject) errors.LintRuleError {
+	// check only `rbac-for-us.yaml` files
+	if !strings.HasSuffix(object.ShortPath(), "rbac-for-us.yaml") {
+		return errors.EmptyRuleError
+	}
+
+	// check Role and ClusterRole for wildcards
+	objectKind := object.Unstructured.GetKind()
+	switch objectKind {
+	case "Role", "ClusterRole":
+		return checkRoles(object)
+	default:
+		return errors.EmptyRuleError
+	}
+}
+
+func checkRoles(object storage.StoreObject) errors.LintRuleError {
+	// check rules for skip
+	for path, rules := range skipCheckWildcards {
+		if strings.EqualFold(object.Path, path) {
+			if slices.Contains(rules, object.Unstructured.GetName()) {
+				return errors.EmptyRuleError
+			}
+		}
+	}
+
+	converter := runtime.DefaultUnstructuredConverter
+
+	role := new(rbac.Role)
+	err := converter.FromUnstructured(object.Unstructured.UnstructuredContent(), role)
+	if err != nil {
+		panic(err)
+	}
+
+	for _, rule := range role.Rules {
+		var objs []string
+		if slices.Contains(rule.APIGroups, "*") {
+			objs = append(objs, "apiGroups")
+		}
+		if slices.Contains(rule.Resources, "*") {
+			objs = append(objs, "resources")
+		}
+		if slices.Contains(rule.Verbs, "*") {
+			objs = append(objs, "verbs")
```

Comment:
Let's separate the errors, please. We can give an advice to a user, how to fix the verbs wildcard
Something like
```
"Verb contains the wildcard. Please, use explicitly set list: `"create", "delete", "deletecollection", "get", "list", "patch", "update", "watch"`
```

Reply:
We can have wildcards in APIGroups, Resources and Verbs. If we separate the errors for each of these groups, should we provide information on the possible values ​​for each of the groups? Or will we do this only for Verbs?
Now we will have only one error for one role indicating in which groups we have wildcards.
If we divide errors into groups, we will have several times more errors. Is it worth doing?